### PR TITLE
Make external tests for MAGIC optional

### LINKED ISSUE #1001: MAGIC in external causes test failures if its not installed
@scottgigante @flying-sheep, on current master the tests fail for me with a MAGIC not installed error. External tests should have some mark to make sure they don't get run if the required package isn't installed.

### PR DESCRIPTION
**Pull request by scottgigante**
Closes #1001

### CODE REVIEW

**Comment by scottgigante** [comment]
Realised there's a better way of doing this with pytest.

File: scanpy/external/pp/_magic.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 from ..._compat import Literal
 
 
+__min_version__ = "2.0"
```
Comment (flying-sheep) [inline scanpy/external/pp/_magic.py]: Don’t use `__dunder__` naming for things that aren’t standardized python names. You can just use `UPPER_WITH_UNDERSCORES` for constants.

File: scanpy/external/pp/_magic.py [hunk 2]
```diff
@@ -136,7 +139,7 @@ def magic(
         )
     else:
         version = tuple([int(v) for v in __version__.split(".")[:2]])
-        min_version = (2, 0)
+        min_version = tuple([int(v) for v in __min_version__.split(".")[:2]])
         if not version >= min_version:
```
Comment (flying-sheep) [inline scanpy/external/pp/_magic.py]: Please compare versions using `packaging.version`

https://github.com/theislab/scanpy/blob/0d44d1d50138e76bdc120cfacfd40c222e47b79e/scanpy/external/tl/_pypairs.py#L150-L152

**Review by flying-sheep — CHANGES_REQUESTED** [review]
Thank you for your quick response. The `tuple` stuff is a bit flaky though, let’s just use standard version parsing and comparing

**Comment by scottgigante** [comment]
Thanks, I hadn't seen `packaging.version` before. Much better!

**Comment by flying-sheep** [comment]
It’s a recent change, driven by the complex stuff people did to compare versions :wink:

**Review by flying-sheep — APPROVED** [review]

**Comment by scottgigante** [comment]
It's not clear to me why these `test_10x` tests are failing here and not on master -- there shouldn't be anything in this diff that affects those tests.

**Comment by flying-sheep** [comment]
You’re right: They would be failing on master if master had been tested since anndata 0.7 was released earlier today. We have to merge #989 to get tests to pass again before we can merge this.

/edit: done, should work now

**Pull request merged**